codec.proto

```diff
@@ -9,304 +9,286 @@ import "google/protobuf/descriptor.proto";
-  Timestamp time = 4 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
-  BlockID last_block_id = 5 [(gogoproto.nullable) = false];
-  bytes last_commit_hash = 6 [(fig.bytetype) = "Hash"];  // hash 256 32 bytes
-  bytes data_hash = 7 [(fig.bytetype) = "Hash"];         // hash 256 32 bytes
-  bytes validators_hash = 8 [(fig.bytetype) = "Hash"];   // hash 256 32 bytes
-  bytes next_validators_hash = 9 [(fig.bytetype) = "Hash"];                        // hash 256 32 bytes
-  bytes consensus_hash = 10 [(fig.bytetype) = "Hash"];  // hash 256 32 bytes
-  bytes app_hash = 11 [(fig.bytetype) = "Hash"];  // determined by application, not a fixed length
-  bytes last_results_hash = 12 [(fig.bytetype) = "Hash"];  // hash 256 32 bytes - first block will be results of an empty hash
-  bytes evidence_hash = 13 [(fig.bytetype) = "Hash"];  // hash sha256 32 bytes
-  Address proposer_address = 14;  // hash 256 20 bytes - this is the first 20 characters of a 32-byte key - SHA256(pubkey)[:20]
+message Duration {
+  int64 seconds = 1;
+  int32 nanos   = 2;
 }
 
-message Consensus {
-  option (gogoproto.equal) = true;
-  uint64 block = 1;
-  uint64 app = 2;
+message DuplicateVoteEvidence {
+  EventVote   vote_a              = 1;
+  EventVote   vote_b              = 2;
+  int64       total_voting_power  = 3;
+  int64       validator_power     = 4;
+  Timestamp   timestamp           = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
 }
 
-message BlockID {
-  bytes hash = 1 [(fig.bytetype) = "Hash"];
-  PartSetHeader part_set_header = 2 [(gogoproto.nullable) = false];
+message Event {
+  string                  type       = 1;
+  repeated EventAttribute attributes = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "attributes,omitempty"
+  ];
 }
 
-message PartSetHeader {
-  uint32 total = 1;
-  bytes hash = 2 [(fig.bytetype) = "Hash"];  // hash 256 32 bytes
+message EventAttribute {
+  bytes key   = 1;
+  bytes value = 2;
+  bool  index = 3; // nondeterministic
 }
 
-message Data {
-  repeated bytes txs = 1;
+message EventBlock {
+  Block               block               = 1;
+  BlockID             block_id            = 2; // not present in v0.34.13
+  ResponseBeginBlock  result_begin_block  = 3;
+  ResponseEndBlock    result_end_block    = 4;
 }
 
-message Evidence {
-  oneof sum {
-    DuplicateVoteEvidence duplicate_vote_evidence = 1;
-    LightClientAttackEvidence light_client_attack_evidence = 2;
-  }
+// EventData
+message EventData {
+  Event     event = 1;
+  EventList block = 2;
 }
 
-message DuplicateVoteEvidence {
-  EventVote vote_a = 1;
-  EventVote vote_b = 2;
-  int64 total_voting_power = 3;
-  int64 validator_power = 4;
-  Timestamp timestamp = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+// EventList
+message EventList {
+  EventBlock                new_block            = 1;
+  repeated EventTx          transaction         = 2;
+  EventValidatorSetUpdates  validator_set_updates = 3;
 }
```

Comment:
I think it makes sense for this to be at the top (after the enums) vs half way through as this is the overall event list that we use and not just another message